rmw_zenoh_cpp/src/detail/rmw_context_impl_s.cpp

```diff
@@ -41,14 +41,24 @@
 // TODO(clalancette): Make this configurable, or get it from the configuration
 #define SHM_BUFFER_SIZE_MB 10
 
+// The variable is used to identify whether the process is trying to exit or not.
+// The atexit function we registered will set the flag and prevent us from closing
+// Zenoh Session. Zenoh API can't be used in atexit function, because Tokio context
+// is already destroyed. It will cause panic if we do so.
+static bool is_exiting = false;
+void update_is_exiting()
+{
+  is_exiting = true;
+}
+
```

Comment:
@Yadunund I created an issue here for further discussion
https://github.com/ros2/rmw_zenoh/issues/324

Reply:
Thanks for clearing explaining the situation in that ticket @evshary!
My preference would be to save this change for a future PR after further consideration and not include it in this PR.

Reply:
Thanks @evshary 

> My preference would be to save this change for a future PR after further consideration and not include it in this PR.

Okay, I will separate it from this PR.

Reply:
Thanks for sharing your thoughts @Yadunund. I agree with you that the migration PR to Zenoh 1.0 should be simpler and it will be easier to review. Let me know if you want me to create a separate PR on the rolling branch or wait until suitable conditions are met.